native/cocos/math/Quaternion.cpp

```diff
@@ -371,45 +371,75 @@ void Quaternion::toEuler(const Quaternion &q, bool outerZ, Vec3 *out) {
 }
 
 void Quaternion::fromMat3(const Mat3 &m, Quaternion *out) {
-    CC_ASSERT(out);
+	CC_ASSERT(out);
     float m00 = m.m[0];
-    float m10 = m.m[1];
-    float m20 = m.m[2];
-    float m01 = m.m[3];
+    float m01 = m.m[1];
+    float m02 = m.m[2];
+    float m10 = m.m[3];
     float m11 = m.m[4];
-    float m21 = m.m[5];
-    float m02 = m.m[6];
-    float m12 = m.m[7];
+    float m12 = m.m[5];
+    float m20 = m.m[6];
+    float m21 = m.m[7];
     float m22 = m.m[8];
-    float trace = m00 + m11 + m22;
-
-    if (trace > 0.F) {
-        const float s = 0.5F / sqrtf(trace + 1.F);
-        out->w = 0.25F / s;
-        out->x = (m21 - m12) * s;
-        out->y = (m02 - m20) * s;
-        out->z = (m10 - m01) * s;
-    } else if ((m00 > m11) && (m00 > m22)) {
-        const float s = 0.5F / sqrtf(1.F + m00 - m11 - m22);
-
-        out->w = (m21 - m12) * s;
-        out->x = 0.25F / s;
-        out->y = (m01 + m10) * s;
-        out->z = (m02 + m20) * s;
-    } else if (m11 > m22) {
-        const float s = 0.5F / sqrtf(1.F + m11 - m00 - m22);
-
-        out->w = (m02 - m20) * s;
-        out->x = (m02 + m10) * s;
-        out->y = 0.25F / s;
-        out->z = (m12 + m21) * s;
-    } else {
-        const float s = 0.5F / sqrtf(1.F + m22 - m00 - m11);
 
-        out->w = (m10 - m01) * s;
-        out->x = (m02 + m20) * s;
-        out->y = (m12 + m21) * s;
-        out->z = 0.25F / s;
+    float fourXSquaredMinus1 = m00 - m11 - m22;
+    float fourYSquaredMinus1 = m11 - m00 - m22;
+    float fourZSquaredMinus1 = m22 - m00 - m11;
+    float fourWSquaredMinus1 = m00 + m11 + m22;
+
+    int biggestIndex = 0;
+    float fourBiggestSquaredMinus1 = fourWSquaredMinus1;
+    if(fourXSquaredMinus1 > fourBiggestSquaredMinus1)
```

Comment:
Should add a space between `if` and `{`.

Reply:
ok